### PR TITLE
Dont send comment reply to user who has community blocked. Fixes #3684

### LINKED ISSUE #3684: [Bug]: Getting comment-reply notifications after blocking a community.
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

Per title.

### Steps to Reproduce

1. Make a comment in a community
2. Block that community
3. Wait until someone replies (or use an alt)
4. You will get notifications

### Technical Details

n/a

### Version

0.18.2

### Lemmy Instance URL

lemmy.world

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

**Comment by Nutomic** [comment]
The same check also needs to be added for blocked instance then.

**Comment by dessalines** [comment]
I thought blocked instance only cares about blocking the communities, not the users from those communities?

**Comment by Nutomic** [comment]
Please add some test cases. Especially for `send_local_notifs()` it would be good to have some test coverage. Maybe as api tests.

File: crates/api_common/src/utils.rs [hunk 1]
```diff
@@ -228,16 +230,42 @@ pub async fn check_person_block(
...
+  person_id: PersonId,
+  pool: &mut DbPool<'_>,
+) -> Result<(), LemmyError> {
+  let is_blocked = InstanceBlock::read(pool, person_id, instance_id).await?;
+  if is_blocked {
+    Err(LemmyErrorType::InstanceIsBlocked)?
+  } else {
+    Ok(())
+  }
+}
```
Comment (Nutomic) [inline crates/api_common/src/utils.rs:267]: These could easily be merged into a single function, no need to do it separately.

File: api_tests/package.json [hunk 2]
```diff
@@ -9,7 +9,12 @@
   "scripts": {
     "lint": "tsc --noEmit && eslint --report-unused-disable-directives --ext .js,.ts,.tsx src && prettier --check 'src/**/*.ts'",
     "fix": "prettier --write src && eslint --fix src",
-    "api-test": "jest -i follow.spec.ts && jest -i post.spec.ts && jest -i comment.spec.ts && jest -i private_message.spec.ts && jest -i user.spec.ts && jest -i community.spec.ts"
+    "api-test": "jest -i follow.spec.ts && jest -i post.spec.ts && jest -i comment.spec.ts && jest -i private_message.spec.ts && jest -i user.spec.ts && jest -i community.spec.ts",
+    "api-test-comment": "jest -i comment.spec.ts",
+    "api-test-post": "jest -i post.spec.ts",
+    "api-test-user": "jest -i user.spec.ts",
+    "api-test-community": "jest -i community.spec.ts",
+    "api-test-private-message": "jest -i private_message.spec.ts"
```
Comment (Nutomic) [inline api_tests/package.json:17]: Nice

File: api_tests/src/comment.spec.ts [hunk 3]
```diff
@@ -740,3 +742,37 @@ test("Report a comment", async () => {
...
+  let commentRes = await createComment(alpha, alphaPost.post.id);
+  expect(commentRes.comment_view.comment.content).toBeDefined();
+  await resolveComment(beta, commentRes.comment_view.comment);
+
+  // Check beta's inbox count, make sure it stays the same
+  let unreadCount = await getUnreadCount(beta);
+  expect(unreadCount.replies).toBe(1);
+
+  let replies = await getReplies(beta);
+  expect(replies.replies.length).toBe(1);
```
Comment (Nutomic) [inline api_tests/src/comment.spec.ts:773]: Should also check these counts before the comment is created to make sure it stays the same. And ensure that `resolveComment` was actually successful.

File: crates/api_common/src/utils.rs (hunk 1 above)
Comment (dessalines) [reply crates/api_common/src/utils.rs:270]: I've made a helper function to combine them, but one of them is used elsewhere individually so I can't combine them all.

File: api_tests/src/comment.spec.ts [hunk 4]
```diff
@@ -763,10 +766,16 @@ test("Dont send a comment reply to a blocked community", async () => {
...
   let commentRes = await createComment(alpha, alphaPost.post.id);
   expect(commentRes.comment_view.comment.content).toBeDefined();
-  await resolveComment(beta, commentRes.comment_view.comment);
+  let alphaComment = await resolveComment(
+    beta,
+    commentRes.comment_view.comment,
+  );
+  if (!alphaComment) {
+    throw "Missing alpha comment before block";
+  }
```
Comment (Nutomic) [inline api_tests/src/comment.spec.ts:775]: I would use simple expect instead of such messages because its shorter and simpler, but oh well

**Review by Nutomic — APPROVED** [review]

**Comment by Nutomic** [comment]
Tests are failing but that seems unrelated.

**Pull request merged**